### ISSUE TITLE
stopEditing() does not take into account the modification?

### ISSUE THREAD

**Issue by ibiza240** [ceolter/ag-grid#1301]
I am using a very simple `BoolCellEditor` and I have a grid with only 1 column editable, so I would like that when I check/uncheck the checkbox, it accepts the modification and stops the editing mode right away.

Currently I have to check the box and press enter to save the change...

I tried putting `this.props.api.stopEditing()` at the end of this function in my `BoolCellEditor`:

    onChangeListener: function(event) {
        var newState = {
            value: event.target.checked
        };
        this.setState(newState);

        // Tried this
        // this.props.api.stopEditing()
    },

and while it does stop the editing, the old value remains as if there was no change.

**Comment by ceolter** [comment]
if you don't want to 'enter and exit edit mode', then don't use a cellEditor, just use a cellRenderer.

**Comment by ibiza240** [comment]
Hi and thanks for your reply,

I would have liked to use the concept of cellValueChanged :

    // in the grid options
    onCellValueChanged: function(event) {
        console.log('onCellValueChanged: ' + event.colDef.field + ' = ' + event.newValue);
    },

Is it still possible when using only a cell renderer?

Thank you! :)

**Comment by ceolter** [comment]
```Is it still possible when using only a cell renderer?```
=> no. but given it's just a checkbox, you should be able to attach your own event simply :)

**Comment by ibiza240** [comment]
Ok this is what I thought, one last question though...is it possible to "pass a function by parameter" to a CellRenderer?

I would like to keep my CellRenderer generic and just execute a provided function when the checkbox changes value.

For now I did it this way (in React)

In the CellRender:

    render: function() {
        return (
            <div className="text-center">
                <input type="checkbox" onChange={this.props.context.onCheckChanged(this.props.value);}/>
            </div>
        );
    },

In my Component containing the CellRenderer, I set the cell gridOptions context to `this` and have the function `onCheckChanged` defined that does the job. But then I have to have a function named exactly `onCheckChanged` in my component and there is a strong tie between the CellRenderer and its caller.

Is there a way to pass an anonymous function to the CellRenderer to execute on value changed or something along those lines that keeps it generic?

Thanks for your time, I hope this all makes sense!
